Code/GraphMol/ChemReactions/ReactionRunner.cpp

```diff
@@ -63,50 +77,52 @@ bool getReactantMatches(const MOL_SPTR_VECT &reactants,
   unsigned int i = 0;
   for (MOL_SPTR_VECT::const_iterator iter = rxn.beginReactantTemplates();
        iter != rxn.endReactantTemplates(); ++iter, i++) {
-    std::vector<MatchVectType> matchesHere;
-    // NOTE that we are *not* uniquifying the results.
-    //   This is because we need multiple matches in reactions. For example,
-    //   The ring-closure coded as:
-    //     [C:1]=[C:2] + [C:3]=[C:4][C:5]=[C:6] ->
-    //     [C:1]1[C:2][C:3][C:4]=[C:5][C:6]1
-    //   should give 4 products here:
-    //     [Cl]C=C + [Br]C=CC=C ->
-    //       [Cl]C1C([Br])C=CCC1
-    //       [Cl]C1CC(Br)C=CC1
-    //       C1C([Br])C=CCC1[Cl]
-    //       C1CC([Br])C=CC1[Cl]
-    //   Yes, in this case there are only 2 unique products, but that's
-    //   a factor of the reactants' symmetry.
-    //
-    //   There's no particularly straightforward way of solving this problem of
-    //   recognizing cases
-    //   where we should give all matches and cases where we shouldn't; it's
-    //   safer to just
-    //   produce everything and let the client deal with uniquifying their
-    //   results.
-    int matchCount = SubstructMatch(*(reactants[i]), *iter->get(), matchesHere,
-                                    false, true, false);
-    BOOST_FOREACH (const MatchVectType &match, matchesHere) {
-      bool keep = true;
-      int pIdx, mIdx;
-      BOOST_FOREACH (boost::tie(pIdx, mIdx), match) {
-        if (reactants[i]->getAtomWithIdx(mIdx)->hasProp(
-                common_properties::_protected)) {
-          keep = false;
-          break;
+    if (matchSingleReactant == MatchAll || matchSingleReactant == i) {
+      std::vector<MatchVectType> matchesHere;
+      // NOTE that we are *not* uniquifying the results.
+      //   This is because we need multiple matches in reactions. For example,
+      //   The ring-closure coded as:
+      //     [C:1]=[C:2] + [C:3]=[C:4][C:5]=[C:6] ->
+      //     [C:1]1[C:2][C:3][C:4]=[C:5][C:6]1
+      //   should give 4 products here:
+      //     [Cl]C=C + [Br]C=CC=C ->
+      //       [Cl]C1C([Br])C=CCC1
+      //       [Cl]C1CC(Br)C=CC1
+      //       C1C([Br])C=CCC1[Cl]
+      //       C1CC([Br])C=CC1[Cl]
+      //   Yes, in this case there are only 2 unique products, but that's
+      //   a factor of the reactants' symmetry.
+      //
+      //   There's no particularly straightforward way of solving this problem
+      //   of recognizing cases
+      //   where we should give all matches and cases where we shouldn't; it's
+      //   safer to just
+      //   produce everything and let the client deal with uniquifying their
+      //   results.
+      int matchCount = SubstructMatch(*(reactants[i]), *iter->get(),
```

Comment:
These aren't changes actually added by this pull request, the differences are indentation (it is now in an if statement).

Perhaps we should make a separate pull request to address these.  

I'm actually surprised we don't see this sign issue with -Wall -Weverything, we have been trying to pay attention to compiler warnings recently.